instrumentation/runtime/go.mod

```diff
@@ -6,13 +6,17 @@ require (
 	github.com/stretchr/testify v1.9.0
 	go.opentelemetry.io/otel v1.28.0
 	go.opentelemetry.io/otel/metric v1.28.0
+	go.opentelemetry.io/otel/sdk v1.28.0
+	go.opentelemetry.io/otel/sdk/metric v1.28.0
```

Comment:
Instrumentation should not depend on the the SDK. This looks to be coming in from the inclusion of the example. I would recommend keeping it as a separate module.

Reply:
This is from the use of metric.Producer.  LMK if you would like the producer in a separate module.

Reply:
I see ...

This is all because we don't have an observable Histogram, right?

Does it make more sense to have all of the runtime metrics come from this `Producer`? That way they are all under the same scope.

Reply:
I considered that.  However, it would mean all other metrics would be pre-aggregated as well. Things like views or temporality configuration would not work unless we added + implemented those features within the producer

Reply:
I could probably use the metrics SDK itself within the producer if we wanted to support those.

Reply:
Hmm, yeah. I'm not really sure which way would be preferable to end-users.

I think in the best case scenario, as an end-user, you would get instrumentation that included all of these metrics (including the `go.schedule.duration`) without having to depend on the SDK. Is this something we need to bring up to the specification? Is adding an observable "pre-binned" histogram instrument the right solution here?

Reply:
i'm honestly not sure what the right solution is.  Having an observable pre-binned histogram instrument (a "bridge" API) would eliminate the need to depend on the SDK here, but it doesn't solve things like views (e.g. you can't change the aggregation to ExponentialHistogram).  We also put producers on the Reader, rather than meter provider, intentionally since producers don't allow exporters to influence aggregations or temporality.

Reply:
> i'm honestly not sure what the right solution is.

Same.

If we were to move everything into the producer, how would you use the SDK to accommodate views? Is the idea to setup some kind of pull exporter and when `Produce` is called pull the SDK data and merge it into our own `metricsdata.ScopeMetrics`?

Reply:
I don't have cycles to take this on this week.  If you are OK with the current implementation, i'd like to start with this.  Having the producer separate means at least our two options (API vs Producer) are at least non-overlapping and give users control

Reply:
SIG notes:

- Go ahead with this approach
- Look into merging scopes in the SDK
- Long term:
  - Metrics bridge API
  - Producer location in the SDK (maybe make it its own package?)

Reply:
I would like to go ahead with this as-is.  I thought about adding an experimental disclaimer, but I didn't want to imply that other parts of the package are non-experimental.  When we stabilize other portions of this package, we can decide if this needs to be moved.